### PR TITLE
Add SSH configuration to snyk workflow

### PR DESCRIPTION
**Pull request by owenwaltonmono**

### CODE REVIEW

File: .github/workflows/snyk_and_licence_check.yml [hunk 1]
```diff
@@ -22,6 +22,16 @@ jobs:
         with:
           python-version: ${{ inputs.python-version }}
           cache: 'pip' # caching pip dependencies
+      
+      # Should consider replacing with
+      # https://github.com/marketplace/actions/webfactory-ssh-agent
+      - name: Setup SSH
+        continue-on-error: true
+        run: |
+          eval `ssh-agent -s`
```
Comment (McPonolith) [inline .github/workflows/snyk_and_licence_check.yml:31]: Why this?

File: .github/workflows/snyk_and_licence_check.yml [hunk 2]
```diff
@@ -22,6 +22,16 @@ jobs:
...
+          echo "SSH_AUTH_SOCK=${SSH_AUTH_SOCK}" >> $GITHUB_ENV
+          ssh-add - <<< "${{ secrets.CI_READ_GH_SSH_KEY }}"
+
```
Comment (McPonolith) [inline .github/workflows/snyk_and_licence_check.yml:34]: I suspect your also going to have to add the host keys as well. take a look at the docker build workflow to see how to setup ssh

**Review by McPonolith — CHANGES_REQUESTED** [review]

File: .github/workflows/snyk_and_licence_check.yml (hunk 1 above)
Comment (owenwaltonmono) [reply .github/workflows/snyk_and_licence_check.yml:31]: I saw that this was being used in other workflow files to add the SSH key to the agent before running other workflows that were running `pip install`.

File: .github/workflows/snyk_and_licence_check.yml (hunk 2 above)
Comment (owenwaltonmono) [reply .github/workflows/snyk_and_licence_check.yml:34]: image

**Pull request merged**